kernel/kernel-impl/src/main/java/org/sakaiproject/authz/impl/DbAuthzGroupSqlDefault.java

```diff
@@ -95,15 +95,15 @@ public String getCountRealmRoleSql()
 		return "select count(1) from SAKAI_REALM_ROLE where ROLE_NAME = ?";
 	}
 	
-	public String getCountRoleFunctionSql(String inClause)
+	public String getCountRoleFunctionSql(String inClause, boolean isDelegated)
 	{
 		return "select count(1) from SAKAI_REALM_RL_FN MAINTABLE "
 				+ "		JOIN SAKAI_REALM_ROLE ROLE ON ROLE.ROLE_KEY = MAINTABLE.ROLE_KEY "
 				+ "		JOIN SAKAI_REALM_FUNCTION FUNCTIONS ON FUNCTIONS.FUNCTION_KEY = MAINTABLE.FUNCTION_KEY "
 				+ "		JOIN SAKAI_REALM REALM ON REALM.REALM_KEY = MAINTABLE.REALM_KEY "
 				+ "		where ROLE.ROLE_NAME = ? AND FUNCTIONS.FUNCTION_NAME = ?"
 				+ "		AND REALM.REALM_KEY in (select REALM_KEY from SAKAI_REALM where " + inClause + ")"

```

Comment:
This sub-select appears completely unnecessary. You already have the SAKAI_REALM table joined.